### PR TITLE
feature/proofpoint_pod_add_fields

### PR DESCRIPTION
**Pull request by clement-burtscher-sekoia**
## Description

Relates to https://github.com/SekoiaLab/integration/issues/1651

### Added

- Parse new custom fields from `msgParts` message:
    - `proofpoint.pod.msgpart.is_deleted`

## Type of Change

- [ ] New intake format
- [X] Update to existing intake format
- [ ] Bug fix
- [ ] Documentation update
- [ ] Other (please describe):

## Affected Intake Formats

- Vendor/product: ProofPoint POD

### 🔒 CRITICAL - Security Requirements (MUST BE VERIFIED FIRST)

- [ ] **No real email addresses** in test files (use: user@example.com, test@example.org)
- [ ] **No real passwords or API keys** (use: "P@ssw0rd!", "xxxxxxxxxxxx", "FAKE_TOKEN_123")
- [ ] **No real IP addresses** (use TEST-NET ranges: 192.0.2.x, 198.51.100.x, 203.0.113.x)
- [ ] **No real phone numbers** (use: +1-555-0100 or similar test numbers)
- [ ] **No real names or PII** (use: "John Doe", "Jane Smith", "User123")
- [ ] **No real company/organization names** in test data (unless public vendors)
- [ ] **No real usernames, hostnames, or domain names** (use: user1, host.example.com, test.corp)
- [ ] All test data has been anonymized and verified

### Required for All PRs

- [ ] Code has been linted
    - [ ] with the linter for manifest, smart-descriptions, and test files (`uv run python linter.py check --changes` in utils/)
    - [ ] with Prettier for the parser code files (`npx prettier --write <module>/<format>/ingest/parser.yml`)
- [ ] All CI/CD checks pass
- [ ] Changes follow the contribution guidelines in `CONTRIBUTING.md`

### Required for New/Updated Intake Formats

- [ ] Clear descriptions provided for modules and formats
- [ ] Logo files included for new modules/formats
- [ ] Parser test coverage is at least 75%
- [ ] At least one smart-description is provided
- [ ] README.md updated (if applicable)
- [ ] Taxonomy mappings are correct and documented
- [ ] Test cases cover edge cases and error handling

### Testing

- [ ] Local tests pass (`uv run pytest` in utils/)
- [ ] Sample data is representative and **completely anonymized**
- [ ] **No sensitive information** (PII, credentials, real IPs, real emails) in test files
- [ ] Test data uses example.com/example.org domains and TEST-NET IP ranges
- [ ] Parser handles malformed data gracefully

### Documentation

- [ ] Code changes are documented
- [ ] Smart-descriptions are clear and informative
- [ ] Field mappings are explained

## Related Issues

Closes #

## Summary by Sourcery

Extend the Proofpoint POD intake format to capture deletion status for message parts.

New Features:
- Add a custom boolean field `proofpoint.pod.msgpart.is_deleted` to represent whether a message part was deleted.

Enhancements:
- Map the Proofpoint `msgParts.isDeleted` attribute into the new `proofpoint.pod.msgpart.is_deleted` field in the parser configuration.

Documentation:
- Update the Proofpoint POD changelog to document the new `isDeleted` message-part field support.

Tests:
- Add test fixtures covering msgParts with deleted flags to validate parsing of the new `is_deleted` field.

### CODE REVIEW

File: ProofPoint/proofpoint-pod/CHANGELOG.md [hunk 1]
```diff
@@ -6,3 +6,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+## [Unreleased] - 2026-06-10
+
+### Added
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply ProofPoint/proofpoint-pod/CHANGELOG.md:12]: Intake-formats changelogs cannot mention the tag (managed by backend) but must mention dates

File: ProofPoint/proofpoint-pod/ingest/parser.yml [hunk 2]
```diff
@@ -129,6 +129,7 @@ stages:
               {% endfor %}
             ]
           proofpoint.pod.msgpart.id: "{{json_event.message.uuid}}"
+          proofpoint.pod.msgpart.is_deleted: "{{json_event.message.msgParts.isDeleted}}"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply ProofPoint/proofpoint-pod/ingest/parser.yml:132]: Currently, ProofPoint POD sends one message part per log as a single object (not an array), so direct access to `msgParts.isDeleted` in the parser is correct; if ProofPoint changed to send multiple parts per log as an array, a loop would be required instead.

File: ProofPoint/proofpoint-pod/CHANGELOG.md [hunk 3]
```diff
@@ -6,3 +6,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+
+- Added parsing of `isDeleted` field from msgParts to `proofpoint.pod.msgpart.is_deleted` custom field
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply ProofPoint/proofpoint-pod/CHANGELOG.md]: Intake-formats changelogs cannot mention the tag (managed by backend) but must mention dates

File: ProofPoint/proofpoint-pod/tests/test_msgparts.json [hunk 4]
```diff
@@ -45,7 +45,8 @@
     "proofpoint": {
       "pod": {
         "msgpart": {
-          "id": "eb99b626-c278-4af3-96f8-5a194e016a43"
+          "id": "eb99b626-c278-4af3-96f8-5a194e016a43",
+          "is_deleted": false
         },
         "urls": [
           "http://docs.yahoo.com/info/terms/",
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply ProofPoint/proofpoint-pod/tests/test_msgparts.json:52]: At first glance, the anonymization check seems too strict for these pre-existing logs, which are still acceptable

File: ProofPoint/proofpoint-pod/tests/test_msgparts_metadata.json [hunk 5]
```diff
@@ -53,7 +53,8 @@
           "modified": "2025-11-20 21:38:41"
         },
         "msgpart": {
-          "id": "11111111-1111-1111-1111-111111111111"
+          "id": "11111111-1111-1111-1111-111111111111",
+          "is_deleted": false
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply ProofPoint/proofpoint-pod/tests/test_msgparts_metadata.json:57]: At first glance, the anonymization check seems too strict for these pre-existing logs, which are still acceptable

**Review by mchupeau-sk — APPROVED** [review]
LGFM
